src/GameLogic/Entities/Fireball.ts

```diff
@@ -64,8 +64,13 @@ export class Fireball implements Entity {
 
 	private move(level: Level, direction: Direction8): void {
 		this.direction = direction;
-		this.x += Direction8Utils.getX(direction);
-		this.y += Direction8Utils.getY(direction);
+		const newX = this.x + Direction8Utils.getX(direction);
+		const newY = this.y + Direction8Utils.getY(direction);
+
+		level.entities.updatePosition(this, newX, newY);
```

Comment:
Is there ever going to be a time where we want to call updatePosition and not also update the x and y value on the entity itself? If not, can udpatePosition set those values?

Alternatively, now we are using a spatial grid, does the entity even need to bother to maintain its own x and y values any more?

Reply:
It's not a bad idea to do the update in the method. I've been trying a lot of different architectural solutions that didn't work out in my opinion and I guess I missed this option :D.

As for maintaining its own x/y, I presume you mean it as "Never set x/y inside the entity" rather than "remove x/y from entity", since the latter is pretty unfeasible. Good idea, I'll make sure that's the case.